fdbserver/workloads/MetaclusterManagementWorkload.actor.cpp

```diff
@@ -771,7 +799,14 @@ struct MetaclusterManagementWorkload : TestWorkload {
 				ASSERT(exists);
 				return Void();
 			} else if (e.code() == error_code_metacluster_no_capacity) {
-				ASSERT(!hasCapacity && !exists);
+				ASSERT(!exists);
+				if (metacluster::AssignClusterAutomatically::False == assignClusterAutomatically) {
+					ASSERT(!hasCapacity);
+				} else {
+					// It's possible that all the data clusters are excluded from the auto-assignment pool.
```

Comment:
Can we validate that this condition is true, potentially by accounting for the auto-assignment behavior in computing `hasCapacity`?

Reply:
Will do in another commit